notify/notify.go

```diff
@@ -525,11 +527,17 @@ func (n *MuteStage) Exec(ctx context.Context, logger log.Logger, alerts ...*type
 	)
 	for _, a := range alerts {
 		// TODO(fabxc): increment total alerts counter.
+		isMuted := n.muter.Mutes(a.Labels)
+		if !isMuted && !a.Resolved() && n.resolveInhibited {
+			a.FiredUnmuted = true
+		}
 		// Do not send the alert if muted.
-		if n.muter.Mutes(a.Labels) {
-			muted = append(muted, a)
-		} else {
+		// Do send resolved notifications if configured
+		// to do so and the alert fired unmuted.
+		if !isMuted || (a.Resolved() && a.FiredUnmuted && n.resolveInhibited) {
 			filtered = append(filtered, a)
+		} else {
+			muted = append(muted, a)
```

Comment:
i actually find this hunk a little confusing. why are we flipping `a.FiredUnmuted here? the conditional looks *really* similar to the second one there, and, what's worse, the `a.FireUnmuted` is reused there which, effectively, makes the second conditional look like:

```golang
!isMuted || (a.Resolved() && (!isMuted && !a.Resolved() && n.resolveInhibited) && n.resolveInhibited) 
```

which i *think* is equivalent to:

```golang
!isMuted || (!a.FiredUnmuted && a.Resolved() && && n.resolveInhibited) 
```

before `a.FiredUnmuted` is reset. i wonder if there's a way to make this code more readable, perhaps with a comment explaing why the `FiredUnmuted` flag is changed there?